src/build/app/mod.rs

```diff
@@ -1601,6 +1855,30 @@ impl<'b> App<'b> {
     fn _debug_asserts(&self) {
         debug!("App::_debug_asserts");
 
+        for sc in &self.subcommands {
+            // Conflicts between flag subcommands and long args
+            if let Some(l) = sc.long {
+                if let Some((first, second)) = self.two_flags_of(|f| f.long() == Some(l)) {
+                    panic!(
+                        "Long option names must be unique for each argument, \
+                        but '--{}' is used by both an {} and an {}",
+                        l, first, second
+                    );
+                }
+            }
+
+            // Conflicts between flag subcommands and short args
+            if let Some(s) = sc.short {
+                if let Some((first, second)) = self.two_flags_of(|f| f.short() == Some(s)) {
```

Comment:
You are not checking the short flag aliases here. If there was a test for it, it would have been caught easily. If you implement long_flag, similarily for the above if condition.

Reply:
Good catch, thanks!